tests/text/test_tabstops.py

```diff
@@ -34,9 +34,30 @@ def it_can_iterate_over_its_tab_stops(self, iter_fixture):
         assert count == expected_count
         assert TabStop_.call_args_list == expected_calls
 
+    def it_can_get_a_tab_stop_by_index(self, index_fixture):
+        tab_stops, expected_vals, tab_stop_, TabStop_, expected_calls = (
+            index_fixture
+        )
+        for count in range(len(tab_stops)):
+            assert tab_stops[count] is tab_stop_
+        assert TabStop_.call_args_list == expected_calls
+
```

Comment:
Alright, here's the fixture for it. I'll go through it line by line:

We start with this minimum known fixture, based on what we found we needed in the test. The return line matches the left-hand-side of the assignment in the test exactly, same spelling and everything:

```python
@pytest.fixture
def index_fixture(self, request):
    return tab_stops, idx, TabStop_, tab, tab_stop_
```

It returns the right items, and is minimal everywhere else, so this is strictly mechanical. This doesn't work because it doesn't have the return values defined anywhere, so we take them one by one, starting with tab_stops. We create an element from some compact XML (cxml) and construct a TabStops object from it. That's step two done.

```python
@pytest.fixture
def index_fixture(self, request):
    pPr = element('w:pPr/w:tabs/w:tab{w:pos=0}')
    tab_stops = TabStops(pPr)
    return tab_stops, idx, TabStop_, tab, tab_stop_
```

For idx, we'll just say 0 here, since there's only one member, that's the only valid value.

```python
@pytest.fixture
def index_fixture(self, request):
    pPr = element('w:pPr/w:tabs/w:tab{w:pos=0}')
    tab_stops = TabStops(pPr)
    idx = 0
    return tab_stops, idx, TabStop_, tab, tab_stop_
```

TabStop_ is a class_mock that's already defined, so we bring it in as a parameter of the fixture. py.test will automatically locate it and pass it into the fixture correctly when it instantiates the fixture.

```python
@pytest.fixture
def index_fixture(self, request, TabStop_):
    pPr = element('w:pPr/w:tabs/w:tab{w:pos=0}')
    tab_stops = TabStops(pPr)
    idx = 0
    return tab_stops, idx, TabStop_, tab, tab_stop_
```

`tab` is that w:tab grandchild element of the w:pPr, so we get that using xPath:

```python
@pytest.fixture
def index_fixture(self, request, TabStop_):
    pPr = element('w:pPr/w:tabs/w:tab{w:pos=0}')
    tab_stops = TabStops(pPr)
    idx = 0
    tab = pPr.xpath('./w:tabs/w:tab')[idx]
    return tab_stops, idx, TabStop_, tab, tab_stop_
```

And tab_stop_ is an instance_mock that's already defined among our fixture components, so we just add that as a parameter as well:

```python
@pytest.fixture
def index_fixture(self, request, TabStop_, tab_stop_):
    pPr = element('w:pPr/w:tabs/w:tab{w:pos=0}')
    tab_stops = TabStops(pPr)
    idx = 0
    tab = pPr.xpath('./w:tabs/w:tab')[idx]
    return tab_stops, idx, TabStop_, tab, tab_stop_
```

Now this one should run fine, but we want a few more cases, so we parameterize the fixture:

```python
@pytest.fixture(params=[
    ('w:pPr/w:tabs/w:tab{w:pos=0}',                                 0),
    ('w:pPr/w:tabs/(w:tab{w:pos=1},w:tab{w:pos=2},w:tab{w:pos=3})', 1),
    ('w:pPr/w:tabs/(w:tab{w:pos=4},w:tab{w:pos=5},w:tab{w:pos=6})', 2),
])
def index_fixture(self, request, TabStop_, tab_stop_):
    pPr_cxml, idx = request.param
    pPr = element(pPr_cxml)
    tab = pPr.xpath('./w:tabs/w:tab')[idx]
    tab_stops = TabStops(pPr)
    return tab_stops, idx, TabStop_, tab, tab_stop_
```

And Bob's your uncle :)